### PR TITLE
Add a test for EtcdDeleteKeys.

### PR DESCRIPTION
**Pull request by pphaneuf**
Also fixed a bug found by the new test, and clarified one of the error cases (giving it a null task). Verified that the bug fixed by #633 would have been exposed by the test.

### CODE REVIEW

File: cpp/util/mock_etcd.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                    const std::chrono::seconds& ttl, Response* resp,
+                    util::Task* task));
+  MOCK_METHOD3(Delete, void(const std::string& key,
+                            const int64_t current_index, util::Task* task));
+  MOCK_METHOD3(Watch, void(const std::string& key, const WatchCallback& cb,
+                           util::Task* task));
+
+  // TODO(pphaneuf): Shouldn't be used, and shouldn't be there, but
+  // has to be for now.
+  MOCK_METHOD5(Generic, void(const std::string& key,
```
Comment (pphaneuf) [inline cpp/util/mock_etcd.h]: Didn't think of it before I sent this PR, but I could make this one private? It needs to be there so that the vtable gets emitted...
